### ISSUE TITLE
CodeNarc output not JSON - Cannot read property 'totalFoundErrorNumber' of undefined

### ISSUE THREAD

**Issue by clifmo** [nvuillam/npm-groovy-lint#107]
Windows 10 

node
```
$ node -v
v12.17.0
```
.groovylintrc.json
```
{
    "extends": "recommended",
    "rules": {
        "LineLength": {
            "enabled": false
        }
    }
}
```

```
$ npm-groovy-lint -p . -f "**/*.groovy" -o "./lintResults.html" -l "info" --failon "error" -c .groovylintrc.json
Unable to run java command: {"status":1,"stdout":"","stderr":"","childJavaProcess":{"_events":{},"_eventsCount":2,"_closesNeeded":1,"_closesGot":1,"connected":false,"signalCode":null,"exitCode":1,"killed":false,"spawnfile":"java","_handle":null,"spawnargs":["java","-Xms256m","-Xmx2048m","-cp","\"C:\\Users\\clif.molina\\AppData\\Roaming\\npm\\node_modules\\npm-groovy-lint\\lib\\java\\CodeNarcServer.jar;C:\\Users\\clif.molina\\AppData\\Roaming\\npm\\node_modules\\npm-groovy-lint\\lib\\java\\*\"","com.nvuillam.CodeNarcServer","--server"],"pid":21728,"stdin":null,"stdout":null,"stderr":null,"stdio":[null,null,null]}}

GroovyLint: Error running CodeNarc:

Unexpected error: Cannot read property 'totalFoundErrorNumber' of undefined
TypeError: Cannot read property 'totalFoundErrorNumber' of undefined
    at NpmGroovyLint.manageReturnCode (C:\Users\clif.molina\AppData\Roaming\npm\node_modules\npm-groovy-lint\lib\groovy-lint.js:482:49)
    at NpmGroovyLint.postProcess (C:\Users\clif.molina\AppData\Roaming\npm\node_modules\npm-groovy-lint\lib\groovy-lint.js:331:14)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async NpmGroovyLint.run (C:\Users\clif.molina\AppData\Roaming\npm\node_modules\npm-groovy-lint\lib\groovy-lint.js:69:13)
    at async C:\Users\clif.molina\AppData\Roaming\npm\node_modules\npm-groovy-lint\lib\index.js:12:9
```

###Update

I saw the issue saying memory allocation was the java error. After fresh boot, it looks like:

```
$  npm-groovy-lint -p . -f "**/*.groovy" -o "./lintResults.html" -l "info" --failon "error" -c .groovylintrc.json
GroovyLint: Successfully processed CodeNarc:
CodeNarc completed: (p1=5; p2=42; p3=1012) 4035ms

Unexpected error: Cannot read property 'totalFoundErrorNumber' of undefined
TypeError: Cannot read property 'totalFoundErrorNumber' of undefined
    at NpmGroovyLint.manageReturnCode (C:\Users\clif.molina\AppData\Roaming\npm\node_modules\npm-groovy-lint\lib\groovy-lint.js:482:49)     
    at NpmGroovyLint.postProcess (C:\Users\clif.molina\AppData\Roaming\npm\node_modules\npm-groovy-lint\lib\groovy-lint.js:331:14)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async NpmGroovyLint.run (C:\Users\clif.molina\AppData\Roaming\npm\node_modules\npm-groovy-lint\lib\groovy-lint.js:69:13)
    at async C:\Users\clif.molina\AppData\Roaming\npm\node_modules\npm-groovy-lint\lib\index.js:12:9

```

###update 2

Installed Java 64-bit... 
Stepping through with debugger:
groovy-lint.js:252 `this.setMethodResult(serverCallResult);`

serverCallResult.codeNarcStdErr: undefined
serverCallResult.codeNarcJsonResult.err: `"Unable to parse CodeNarc completed: (p1=5; p2=42; p3=1012) 1606ms\r\n"`
serverCallResult.codeNarcStdOut: `'CodeNarc completed: (p1=5; p2=42; p3=1012) 1606ms\r\n'`

Update 3:

Tracking it to codenarc-caller.js:353 `return JSON.parse(jsonLine);`

The output lines are not JSON and getCodeNarcJsonResult. There are two lines in the response,  `'CodeNarc completed: (p1=5; p2=42; p3=1012) 1606ms\r\n'` and a blank line.

I see where your [PR](https://github.com/CodeNarc/CodeNarc/pull/485) was accepted to add JSON output format to CodeNarc. I see v1.6.1 of CodeNarc is contained in this package.

**Comment by nvuillam** [comment]
@clifmo do you still have the issue ?

If yes, please can you see if you still have it with the beta of next major version ? npm i npm-groovy-lint@beta

Thanks ! :)

**Comment by clifmo** [comment]
Hello,

It's been a little while, so I don't remember what classes I ran it against. However, I am no longer seeing the JSON issue as before. Thanks!

**Comment by nvuillam** [comment]
Great, thanks for the feedback :)

**Comment by cvakiitho** [comment]
I'm getting the same issue with latest npm-groovy-lint:

```

npx npm-groovy-lint --ignorepattern '**/bin/**/*' --verbose --noserver --failonerror

npx: installed 75 in 4.088s

GroovyLint: Error running CodeNarc: 

Picked up JAVA_TOOL_OPTIONS: -XX:+UseG1GC

java.lang.ArrayIndexOutOfBoundsException: -2

	at java.util.ArrayList.elementData(ArrayList.java:424)

	at java.util.ArrayList.get(ArrayList.java:437)

	at org.codenarc.rule.AbstractAstVisitor.sourceLine(AbstractAstVisitor.java:72)

	at sun.reflect.GeneratedMethodAccessor107.invoke(Unknown Source)

	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)

	at java.lang.reflect.Method.invoke(Method.java:498)

	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)

	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)

	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:351)

	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.callCurrent(PogoMetaClassSite.java:61)

	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:185)

	at org.codenarc.rule.formatting.IndentationAstVisitor$_recordMethodColumnAndSourceLineForClosureBlocks_closure7.doCall(IndentationRule.groovy:233)

	at sun.reflect.GeneratedMethodAccessor264.invoke(Unknown Source)

	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)

	at java.lang.reflect.Method.invoke(Method.java:498)

	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)

	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)

	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:263)

	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1026)

	at groovy.lang.Closure.call(Closure.java:412)

	at groovy.lang.Closure.call(Closure.java:428)

	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2318)

	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2303)

	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2344)

	at org.codehaus.groovy.runtime.dgm$200.invoke(Unknown Source)

	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoMetaMethodSiteNoUnwrapNoCoerce.invoke(PojoMetaMethodSite.java:247)

	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:56)

	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:139)

	at org.codenarc.rule.formatting.IndentationAstVisitor.recordMethodColumnAndSourceLineForClosureBlocks(IndentationRule.groovy:230)

	at sun.reflect.GeneratedMethodAccessor268.invoke(Unknown Source)

	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccesso
```

seems to be working fine on 7.6.4
